File: clients/php/src/Viser/Api/REST/DashboardClient.php
<?php

namespace Viser\Api\REST;

use Guzzle\Http\ClientInterface;
use Guzzle\Http\Exception\BadResponseException;
use Viser\Api\REST\Exception\ResponseException;
use Viser\Api\REST\Exception\UnknownException;

class DashboardClient
{
    /**
     * @var ClientInterface
     */
    protected $client;

    /**
     * @var string
     */
    protected $path;

    /**
     * @var string
     */
    protected $version;

    /**
     * @param ClientInterface $client
     */
    public function __construct(ClientInterface $client, $path, $version)
    {
        $this->client = $client;
        $this->path = $path;
        $this->version = $version;
    }

    /**
     * @param string $source
     * @param string $target
     *
     * @return void
     */
    public function event($source, $target)
    {
        $headers = [
            'Content-Type' => 'application/json'
        ];

        $body = new \stdClass();
        $body->source = $source;
        $body->target = $target;

        try {
            $request = $this->client->post(
                $this->path . '/' . $this->version . '/event',
                $headers,
                \json_encode($body)
            );
            $request->send();
        } catch (BadResponseException $e) {
            $response = $e->getResponse();
            $code = $response->getStatusCode();
            $message = $response->getReasonPhrase();
            throw new ResponseException($message, $code);
        } catch (\Exception $e) {
            throw new UnknownException($e->getMessage(), $e->getCode());
        }
    }
}

File: clients/java/src/main/java/com/github/viser/api/rest/DashboardClient.java
package com.github.viser.api.rest;

import com.github.viser.api.rest.exception.ResponseException;
import com.github.viser.api.rest.exception.UnknownErrorException;
import com.github.viser.api.rest.request.EventRequest;
import com.github.viser.api.rest.response.EventResponse;

import javax.ws.rs.BadRequestException;
import javax.ws.rs.client.Entity;
import javax.ws.rs.client.WebTarget;
import javax.ws.rs.core.MediaType;

/**
 * Simple Viser dashboard REST client.
 */
public class DashboardClient
{
    private WebTarget webTarget;

    public DashboardClient(WebTarget webTarget)
    {
        this.webTarget = webTarget;
    }

    /**
     * Triggers an event with default empty ID in Viser dashboard.
     *
     * @param source start of an event
     * @param target end of an event
     *
     * @throws com.github.viser.api.rest.exception.ResponseException
     * @throws com.github.viser.api.rest.exception.UnknownErrorException
     */
    public void event(String source, String target)
    {
        event("", source, target);
    }

    /**
     * Triggers an event in Viser dashboard.
     *
     * @param id     client ID
     * @param source start of an event
     * @param target end of an event
     *
     * @throws com.github.viser.api.rest.exception.ResponseException
     * @throws com.github.viser.api.rest.exception.UnknownErrorException
     */
    public void event(String id, String source, String target)
    {
        try {
            webTarget
                    .request(MediaType.APPLICATION_JSON_TYPE)
                    .post(createEventRequestEntity(id, source, target), EventResponse.class);
        } catch (BadRequestException e) {
            throw new ResponseException();
        } catch (Exception e) {
            throw new UnknownErrorException();
        }
    }

    public Entity<EventRequest> createEventRequestEntity(String id, String source, String target)
    {
        return Entity.json(new EventRequest(id, source, target));
    }
}
